klangbecken/playlist.py

```diff
@@ -158,6 +153,46 @@ def ffmpeg_audio_analyzer(playlist, fileId, ext, file_):
             f"Check your file."
         )
 
+    return cue_in, cue_out
+
+
+def ffmpeg_audio_analyzer(playlist, fileId, ext, file_):
+    """Analyze the audio data with ffmpeg.
+
+    This function does three things:
+    - Check the encoding quality of the track (bitrate, samplerate, stereo/mono)
+    - Calculate the replaygain value for loudness normalization.
+    - Detect silence periods at the start and end of the track, and
+      calculate the according cue points.
+    """
+    # To make sure that we find the correct cue_out point, we append ~0.2 seconds
+    # of silence to the end of the track (with the `apad=pad_len=10000` option).
+    # This guarantees that we always find at least one silence period.
+    #
+    # Also, we have no a priori knowledge of the exact length of the track, to which
+    # we could fall back to. Whereas at the start of the track it is easy: we can
+    # always fall back to 0.0.
+    command = """ffmpeg -i - -af
```

Comment:
you might at some point want to consider using bindings like `ffmpeg-python`, it would get rid of quite a few regexes in this file

Reply:
Hm, `ffmpeg-python` does not support getting back messages from `replaygain` and `silencedetect` as far as I can see. Too bad.

Reply:
:(